src/strings.js

```diff
@@ -40,54 +40,112 @@ function rewritePatternForDotAll(pattern) {
   return cachedRegExp[pattern];
 }
 
-engine.indexOf = function(coll, substr){
+engine.indexOf = function (coll, substr) {
   const str = misc.singleton(coll, 'String');
   return util.isEmpty(substr) || util.isEmpty(str) ? [] : str.indexOf(substr);
 };
 
-engine.substring = function(coll, start, length){
+engine.substring = function (coll, start, length) {
   const str = misc.singleton(coll, 'String');
   if (util.isEmpty(str) || util.isEmpty(start) || start < 0 || start >= str.length) {
-    return  [];
+    return [];
   }
   if (length === undefined || util.isEmpty(length)) {
     return str.substring(start);
   }
   return str.substring(start, start + length);
 };
 
-engine.startsWith = function(coll, prefix){
+engine.startsWith = function (coll, prefix) {
   const str = misc.singleton(coll, 'String');
   return util.isEmpty(prefix) || util.isEmpty(str) ? [] : str.startsWith(prefix);
 };
 
-engine.endsWith = function(coll, postfix) {
+engine.endsWith = function (coll, postfix) {
   const str = misc.singleton(coll, 'String');
   return util.isEmpty(postfix) || util.isEmpty(str) ? [] : str.endsWith(postfix);
 };
 
-engine.containsFn = function(coll, substr){
+engine.containsFn = function (coll, substr) {
   const str = misc.singleton(coll, 'String');
   return util.isEmpty(substr) || util.isEmpty(str) ? [] : str.includes(substr);
 };
 
-engine.upper = function(coll){
+engine.upper = function (coll) {
   const str = misc.singleton(coll, 'String');
   return util.isEmpty(str) ? [] : str.toUpperCase();
 };
 
-
-engine.lower = function(coll){
+engine.lower = function (coll) {
   const str = misc.singleton(coll, 'String');
   return util.isEmpty(str) ? [] : str.toLowerCase();
 };
 
+engine.joinFn = function (coll, separator) {
+  const stringValues = coll.map((n) => {
+    const d = util.valData(n);
+    if (typeof d === "string") {
+      return d;
+    }
+    return null;
+  });
+  return stringValues.join(separator);
+};
+
+engine.splitFn = function (coll, separator) {
+  const strToSplit = misc.singleton(coll, 'String');
+  return util.isEmpty(strToSplit) ? [] : strToSplit.split(separator).filter((v) => { return v != ""; });
```

Comment:
This change was done based on the feedback that you both provided over in the other PR.
(and unit test to catch it too)

Reply:
It would have made this discussion easier if you had finished your other PR rather than starting a conversation there and continuing it here.

Looking again back at the other PR, I don't see any place in the other PR's comments where I said anything about split().  I see Yury commented about split, but not about dropping values that are the empty string.  Anyway, the real question is whether it is the right thing to do.   I think there are probably cases where it isn't.

Reply:
In a previous PR, my comment about the `split()` function was on the wrong line. I mixed up the `trimFn` and `joinFn` functions. I thought Paul commented `joinFn`. They were too similar. Sorry for confusing everyone.

Reply:
I mean this comment: https://github.com/HL7/fhirpath.js/pull/122/files#r975665744

Reply:
I've further noted this discussion on Zulip here:
https://chat.fhir.org/#narrow/stream/179266-fhirpath/topic/split.20.2F.20trim
And logged a ticket for clarification in the spec.
https://jira.hl7.org/browse/FHIR-41033

Reply:
Have removed the filtering of empty strings, will need to revert if the change goes the other way - doubt it though.